processor/filterprocessor/telemetry.go

```diff
@@ -0,0 +1,127 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package filterprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.uber.org/multierr"
+)
+
+const (
+	processorKey = "filter_processor"
+	scopeName    = "go.opentelemetry.io/collector/processor/filterprocessor"
+)
+
+var (
+	processorTagKey      = tag.MustNewKey(processorKey)
+	statMetricsFiltered  = stats.Int64("metrics_filtered", "Number of metrics dropped by the filter processor", stats.UnitDimensionless)
+	statMetricsProcessed = stats.Int64("metrics_processed", "Number of metrics processed by the filter processor", stats.UnitDimensionless)
```

Comment:
I think we should use `.` instead of `_` to namespace the metric names (some other components don't do this yet bc they were created before that naming was determined but I believe we're in the middle of updating them).